lib/Backends/CPU/CPUDeviceManager.cpp

```diff
@@ -50,6 +51,12 @@ void CPUDeviceManager::addNetworkImpl(const Module *module,
       readyCB(module, ResultCode::Failed);
       return;
     }
+
+    if (func.second->getCompileBackendKind() != BackendKind::CPU) {
```

Comment:
Minor nit: This check is almost identical for all device managers. May be it can be moved into a utility method or into a base class method?

Reply:
I could add a helper method `bool checkBackendKind(CompiledFunction*)` but it would replace just the check unless we wanted to pass in all the local state (cb etc). It would end up adding code not removing code, do you think it's clearer?

Reply:
OK. In this case let's leave it as is.